### ISSUE TITLE
rdf.list is not a function

### ISSUE THREAD

**Issue by hunterlester** [maidsafe/safe_app_nodejs#323]
So far I don't see a `list` operation exposed in [rdflib.js](https://github.com/linkeddata/rdflib.js/blob/master/src/index.js)

**Comment by hunterlester** [comment]
The purpose of this operation is to add an ordered list, collection, of nodes to a graph.
It may have been replaced by [rdf.collection](https://github.com/linkeddata/rdflib.js/blob/master/src/data-factory.js#L69).

**Comment by b-zee** [comment]
This issue, and the two other issues #316 and #317, kind of worries me. I [mentioned this worry](https://forum.safedev.org/t/prerfc-data-api-ideas/2069/2?u=bzee) (second and third paragraph) on the forum before.

Is there a good reason to not expose the `rdflib.js` functions directly? Why is it needed to proxy/mirror all these functions from `rdflib.js`?

**Comment by b-zee** [comment]
I guess my worry stems from Don't Repeat Yourself (DRY). These functions cause technical debt in the sense that they require close monitoring by every update of rdflib.js, which is now still at version zero so very prone to changes. (Function name changes, argument changes, new functions, functions removed, etc.)

Developers can also directly use the rdflib.js through the `graphStore` property. In object oriented programming I guess a solution could be to make the object extend the graph though, but JavaScript is not that easy in that regard.

In short: this requires manual work to keep the API aligned with its dependency.

**Comment by bochaco** [comment]
Right, yes I agree, another way would be to have our RDF emulation to simply expose functions to parse, serialise and commit (which will do the internal serialisation to store grpahs in the network), and the leave the rest out so the user decides which lib to use to manipulate grpahs in memory before they are parsed by our RDF emulation. 

In any case, there is some ongoing effort also from @nbaksalyar to see how to expose such an API from safe_client_libs so we can then make it available in all languages and not only JS, and there is also a proposal from @joshuef (https://forum.safedev.org/t/javascript-rdf-tooling-ideas/2239) for alternatives to all this, it's a big topic actually.
